src/lib/features/feature-toggle/feature-toggle-strategies-store.ts

```diff
@@ -149,6 +150,30 @@ function mapStrategyUpdate(
     return update;
 }
 
+function mergeAll<T>(objects: Partial<T>[]): T {
+    return merge.all<T>(objects.filter((i) => i));
+}
+
+const defaultParameters = (
+    params: PartialSome<IFeatureStrategy, 'id' | 'createdAt'>,
+) => {
+    if (params.strategyName === 'gradualRollout') {
+        return {
+            rollout: '100',
+            stickiness: 'default',
+            groupId: params.featureName,
+        };
+    } else {
+        /// We don't really have good defaults for the other kinds of known strategies, so return an empty map.
```

Comment:
Honestly it might be worth returning an error at this point. There's no way you can provide sane defaults for something like userWithIdStrategy, but that is definitely broken without config

I think this probably covered 99% of stuff though. Gut feeling is most people use either the default gradual rollout for most things and standard for some edge cases (last 1% is weird things like IP address strategies but those should fail to create without config)

Reply:
Is `{ "userIds": "" }` any better than it just being null, how does the frontend handle that?

Reply:
No idea how the front end handles parameters being undefined for user id strategy, but having no user IDs specified is totally valid 💁🏼

Let me see what happens if I create it via the API.

Reply:
The UI doesn't appear to care in this case. It treats an empty parameters the same as having an empty list of user ids. This is good in the functional sense, but we could still fix the data for the strategy before inserting it 🤷🏼  I dunno. Doesn't matter much for this, anyway (because it works), but it feels like a better thing(TM) to store something with the required configuration data than to just assume that any clients will auto-fill it later.